PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -156,6 +165,30 @@ function transformReports(inputReportFiles: string[], index: number)
     }
 }
 
+function mapToAnalyzer(node: any) {
+    let ruleId = node.attributes.rule;
+    let violationType = node.name;
+    if (!ruleAnalyzerPairs.get(ruleId)) {
+        switch (violationType) {
+            case 'DupViol':
+                ruleAnalyzerPairs.set(ruleId, "com.parasoft.xtest.cpp.analyzer.static.dupcode");
+                break;
+            case 'FlowViol':
+                ruleAnalyzerPairs.set(ruleId, "com.parasoft.xtest.cpp.analyzer.static.flow");
+                break;
+            case 'MetViol':
+                ruleAnalyzerPairs.set(ruleId, "com.parasoft.xtest.cpp.analyzer.static.metrics");
+                break;
+            case 'StdViol':
+                if (node.attributes.cat == 'GLOBAL') {
```

Comment:
In my legacy report, "cat" attribute is generated for violation elements. I guess that's why we assume we can use "cat" here.
However, this attribute is optional in the schema.
While for "Rule" element, "cat" is a required attribute. 
I suggest that while looping through all rules, if it's for legacy report, we should generate a set which only stores rule id that has "cat" equals to "GLOBAL" and change the condition here.
 

@rwang-parasoft  If you are agree with me and we think it's not quick fix, we can merge this pull request first for other developer's convenience and send another pull request for the change.